ios/Exponent/ExpoKit/ExpoKit.m

```diff
@@ -107,11 +111,13 @@ - (void)setUpApplication:(UIApplication *)application withLaunchOptions:(nullabl
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
   if (![UNUserNotificationCenter currentNotificationCenter].delegate) {
-    [[UNUserNotificationCenter currentNotificationCenter] setDelegate:(id<UNUserNotificationCenterDelegate>) [EXKernel sharedInstance].serviceRegistry.notificationsManager];
-    // This is safe to call; if the app doesn't have permission to display user-facing notifications
-    // then registering for a push token is a no-op
-    [[EXKernel sharedInstance].serviceRegistry.remoteNotificationManager registerForRemoteNotifications];
+    UMLogWarn(@"UNUserNotificationCenter delegates should be set by EXNotificationCenterDelegate.");
+    return YES;
```

Comment:
Maybe not-returning here would make it more error-proof?

I mean, if eg. `priority` sorting didn't work it would be possible that we would end up in a situation where this would be called before `EXNotificationCenterDelegate`'s `application:didFinishLaunchingWithOptions`. If we register the `notificationsManager` as a delegate of `EXNotificationCenterDelegate` nonetheless, everything would work as it should (even though the delegates sorting stopped working). What do you think?